draft-ietf-quic-transport.md

```diff
@@ -1857,15 +1862,22 @@ congestion state (see {{migration-cc}}), so the port SHOULD only be changed
 infrequently.
 
 An endpoint that receives a successfully authenticated packet with a previously
-unused connection ID MUST use a new connection ID for any future packets it
-sends to that address.  To avoid changing connection IDs multiple times when
-packets arrive out of order, endpoints MUST change only in response to a packet
-that increases the largest received packet number.  Failing to do this could
-allow for use of that connection ID to link activity on new paths.  There is no
-need to move to a new connection ID if the address of a peer changes without
-also changing the connection ID.  If no new connection IDs are available, the
+unused connection ID MUST advance to the connection ID whose sequence number is
+equal to the sequence number of the newly received connection ID for any future
+packets it sends to that address.  If that connection ID cannot be used (e.g.,
+it was used on a different path or has not been received), it MUST advance to a
+connection ID with a greater sequence number for any future packets it sends to
+that address.  Failing to do this could allow for use of that connection ID to
+link activity on new paths. There is no need to move to a new connection ID if
+the address of a peer changes without also changing the connection ID, or if the
+connection ID changes to the one with the sequence number equal to the
+endpoint's current connection ID.  If no new connection IDs are available, the
 endpoint MUST NOT send additional packets until a NEW_CONNECTION_ID frame is
-received.
+received.  However, an implementation need not stop sending packets due to a gap
+in sequence numbers; it simply advances to the next currently-available
+connection ID it has received.  Connection IDs with earlier sequence numbers
+which arrive later MAY be retained for use on other local addresses or
+discarded.
```

Comment:
OMG wall of text :)

When this happens, it's probably cause to create a new section.

FWIW, I like the idea that once you use a connection ID on a given interface, you can't use a lower-numbered one.  That could extend to responses.  Once you see a connection ID on a given tuple, then you can't use a lower-numbered one.  That leads to some simple rules:

When you use a new interface/address you MUST use of a new connection ID.
When you use a new connection ID, you MUST use one with a higher sequence number than you have ever used before.
You MUST NOT use a connection ID with a lower sequence number than you have either sent or received on that path.

Formulating the text around that is what we need.